File: src/Assignment1.java
import java.util.Scanner;

class Assignment1 {
    public static void main(String[] args) {
        final Scanner s = new Scanner(System.in);

        System.out.println("Please enter the course name.");

        final String n = s.nextLine();

        System.out.println("Please enter the average time spent in a week for this course in minutes.");

        int time = s.nextInt();

        System.out.println("Please enter the homework grades for this course.");

        final int homework1 = s.nextInt();
        final int homework2 = s.nextInt();
        final int homework3 = s.nextInt();
        final int homework4 = s.nextInt();

        System.out.println("Please enter the quiz grades for this course.");

        final double quiz1 = s.nextDouble();
        final double quiz2 = s.nextDouble();

        System.out.println("Please enter the final exam grade for this course.");

        final double exam = s.nextDouble();

        s.close();

        System.out.println(String.format("Course name: %s", n));

        final int minutes = time % 60;

        int hours = 0;

        while (time > 60) {
            time -= 60;
            hours += 1;
        }

        double homeworkAverage = ((double) (homework1 + homework2 + homework3 + homework4)) / 4;
        double quizAverage = (quiz1 + quiz2) / 2;

        System.out.println(String.format("Weekly time spent: %dh%d", hours, minutes));

        System.out.println("Average homework grade: " + homeworkAverage);

        System.out.println("Average quiz grade: " + quizAverage);

        System.out.println("Final exam grade: " + exam);

        double homeworkWeight = 0.35;
        double quizWeight = 0.15;
        double examWeight = 0.50;

        double average = ((homeworkWeight * homeworkAverage) + (quizWeight * quizAverage) + (exam * examWeight));

        System.out.println("Overall grade: " + Assignment1.round(average));

    }

    public static int round(double n) {
        int z = (int) n;

        if ((n - z) > 0.5) {
            return z + 1;
        }

        return z;
    }
};

File: rust/src/assignment1.rs
use crate::tools::Scanner;

pub fn assignment() {
    let s = Scanner::new();

    writeln!(f, "Please enter the course name.");
    let name = s.next_line();

    writeln!(f, "Please enter the average time spent in a week for this course in minutes.");

    let mut time = s.next_i32();

    writeln!(f, "Please enter the homework grades for this course.");

    let homework1 = s.next_i32();
    let homework2 = s.next_i32();
    let homework3 = s.next_i32();
    let homework4 = s.next_i32();

    writeln!(f, "Please enter the quiz grades for this course.");

    let quiz1 = s.next_double();
    let quiz2 = s.next_double();

    writeln!(f, "Please enter the final exam grade for this course.");

    let exam = s.next_double();

    writeln!(f, "Course name: {name}");

    let minutes = time % 60;
    let mut hours = 0;

    while time > 60 {
        time -= 60;
        hours += 1;
    }

    let homework_average = ((homework1 + homework2 + homework3 + homework4) as f64) / 4.0;
    let quiz_average = (quiz1 + quiz2) / 2.0;

    writeln!(f, "Weekly time spent: {hours}h{minutes}");
    writeln!(f, "Average homework grade: {homework_average}");
    writeln!(f, "Average quiz grade: {quiz_average}");
    writeln!(f, "Final exam grade: {exam}");

    let homework_weight = 0.35;
    let quiz_weight = 0.15;
    let exam_weight = 0.50;

    let average = (homework_weight * homework_average)
        + (quiz_weight * quiz_average)
        + (exam_weight * exam);

        writeln!(f, "Overall grade: {}", average.round());
}
